autopush/utils.py

```diff
@@ -401,32 +452,41 @@ def from_message_table(cls, uaid, item):
         # type: (uuid.UUID, Union[Dict[str, Any], Item]) -> WebPushNotification
         """Create a WebPushNotification from a message table item"""
         key_info = cls.parse_sort_key(item["chidmessageid"])
-        if key_info.get("topic"):
+        if key_info["api_ver"] in ["01", "02"]:
             key_info["message_id"] = item["updateid"]
 
-        return cls(uaid=uaid,
-                   channel_id=uuid.UUID(key_info["channel_id"]),
-                   data=item.get("data"),
-                   headers=item.get("headers"),
-                   ttl=item["ttl"],
-                   topic=key_info.get("topic"),
-                   message_id=key_info["message_id"],
-                   update_id=item.get("updateid"),
-                   timestamp=item.get("timestamp"),
-                   )
+        notif = cls(
+            uaid=uaid,
+            channel_id=uuid.UUID(key_info["channel_id"]),
+            data=item.get("data"),
+            headers=item.get("headers"),
+            ttl=item["ttl"],
+            topic=key_info.get("topic"),
+            message_id=key_info["message_id"],
+            update_id=item.get("updateid"),
+            timestamp=item.get("timestamp"),
+            sortkey_timestamp=key_info.get("sortkey_timestamp")
+        )
+
+        # Ensure we generate the sort-key properly for legacy messges
+        if key_info["api_ver"] == "00":
```

Comment:
this could go into the constructor call too

legacy=key_info["api_ver"] == "00"